Add information on proxy deposits (#976)
w3f/polkadot-wiki: docs/learn-proxies.md

By creating multiple accounts that act for a single account, it lets you come up with more granular
security practices around how you protect private keys while still being able to actively
participate in a network.

## Proxy Deposits

Proxies require deposits in the native currency (i.e. DOT or KSM) in order to be created. The
deposit is required because adding a proxy requires some storage space on-chain, which must be
replicated across every peer in the network. Due to the costly nature of this, these functions could
open up the network to a Denial-of-Service attack. In order to defend against this attack, proxies

> require a deposit to be locked while the storage space is consumed over the life time of the proxy.

Reviewer: ```suggestion
require a deposit to be reserved while the storage space is consumed over the life time of the proxy.
```

When the proxy is removed, so is the storage space, and therefore the deposit is returned.

The deposits are calculated in the runtime, and the function can be found

> [in the runtime code](https://github.com/paritytech/polkadot/blob/e75ae5b712ea4faffce753a04d81b636a22066d0/runtime/polkadot/src/lib.rs#L751).

Reviewer 2: Let's not link to code, despite it being a specific commit. It's subject to change and one more thing to keep an eye on

For example, the deposits are calculated in Polkadot with the following functions:

```rust
// One storage item; key size 32, value size 8; .
pub const ProxyDepositBase: Balance = deposit(1, 8);
// Additional storage item size of 33 bytes.
pub const ProxyDepositFactor: Balance = deposit(0, 33);
```

> The `ProxyDepositBase` is the required amount to be locked up for an account to have a proxy list.

Reviewer: The proxyDepositBase is cost to put the key into storage then the `ProxyDepositFactor is the value associated with the key is an array, so you have to pay a little extra for each item in the array. I would just add these in to make it more clearer to the reader

> For every proxy the account has an additonal amount defined by the `ProxyDepositFactor` is locked up

Reviewer: It would be good to mention here that this reserved fee is returned if you close the proxy

Reviewer: Oh wait I see it above =)

> For every proxy the account has an additonal amount defined by the `ProxyDepositFactor` is locked up

Reviewer: ```suggestion
For every proxy the account has an additonal amount defined by the `ProxyDepositFactor` is reserved
```
I would change this word to reserved because the function in the codebase is named reserve

> For every proxy the account has an additonal amount defined by the `ProxyDepositFactor` is locked up

Reviewer 2: ```suggestion
For every proxy the account has, an additonal amount defined by the `ProxyDepositFactor` is locked up
```
